File: worblehat-domain/src/main/java/de/codecentric/psd/worblehat/domain/BookService.java
package de.codecentric.psd.worblehat.domain;

import javax.annotation.Nonnull;
import java.util.List;
import java.util.Optional;
import java.util.Set;

/**
 * The interface of the domain service for books.
 */
public interface BookService {

	void returnAllBooksByBorrower(String string);

	void returnBookByBorrowerAndIsbn(String borrower, String isbn);

	void returnBookByBorrowerAndTitle(String borrower, String title);

	Optional<Borrowing> borrowBook(String isbn, String borrower);

	Set<Book> findBooksByIsbn(String isbn);

	List<Book> findAllBooks();

	List<Borrowing> findBooksByEmail(String email);

	Optional<Book> createBook(String title, String author, String edition, String isbn, int yearOfPublication);

	boolean bookExists(String isbn);

	void deleteAllBooks();
}

File: worblehat-domain/src/main/java/de/codecentric/psd/worblehat/domain/BookService.kt
package de.codecentric.psd.worblehat.domain

import java.util.*

/**
 * The interface of the domain service for books.
 */
interface BookService {

    fun returnAllBooksByBorrower(borrowerEmailAddress: String)

    fun returnBookByBorrowerAndIsbn(borrower: String, isbn: String)

    fun returnBookByBorrowerAndTitle(borrower: String, title: String)

    fun borrowBook(isbn: String, borrower: String): Optional<Borrowing>

    fun findBooksByIsbn(isbn: String): Set<Book>

    fun findAllBooks(): List<Book>

    fun findBooksByEmail(email: String): List<Borrowing>

    fun createBook(title: String, author: String, edition: String, isbn: String, yearOfPublication: Int): Optional<Book>
    fun createBook(title: String, author: String, edition: String, isbn: String, yearOfPublication: Int, description: String): Optional<Book>

    fun bookExists(isbn: String): Boolean

    fun deleteAllBooks()
}
